lib/trip_dispatcher.rb

```diff
@@ -90,6 +93,58 @@ def load_trips
       trips
     end
 
+    # addition that takes passenger_id to create new trip
+    def request_trip(passenger_id)
+      check_id(passenger_id)
+      id = trips.length + 1
+      available_driver = find_least_recently_active_driver
+      return if available_driver.nil?  # nil return for no available drivers
+      # can be used with passenger in future method to indicate that no trip was
+      # created because no drivers are available
+      passenger = find_passenger(passenger_id)
+      new_trip = {
+        id: id,
+        driver: available_driver,
+        passenger: passenger,
+        start_time: Time.now,
+        end_time: nil,
+        cost: nil,
+        rating: nil
+      }
+
+      trips << trip = Trip.new(new_trip)
+      available_driver.start_new_trip(trip)
+      passenger.add_trip(trip)
+      trip
+    end
+
+    # finds first driver with no ride history and then driver that is least
+    # recently active
+    # if no driver is available returns nil
+    def find_least_recently_active_driver
+      recent_rides = []
+      @drivers.find_all do |driver|
+        if driver.status == :AVAILABLE && driver.trips.empty?
+          return driver
+        elsif driver.status == :AVAILABLE
+          available_drivers = []
+          unsorted_rides = []
+          available_drivers << driver
```

Comment:
look at `available_drivers` you're setting it to an empty array and then inserting 1 driver.